File: app/src/main/java/com/partycalc/database/ActivePartiesDAO.java
package com.partycalc.database;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

@Dao
public interface ActivePartiesDAO {

    @Insert
    long[] insert(ActiveParties... item);

    @Update
    void update(ActiveParties... item);

    @Delete
    void delete(ActiveParties... item);

    @Query("SELECT participant.* from activeparties " +
            "LEFT JOIN participant ON participant.id=activeparties.participantId " +
            "where partyId=:partyId")
    LiveData<List<Participant>> getParticipantsForParty(final int partyId);

    @Query("delete from ActiveParties where participantId=:participantId and partyId=:partyId")
    void removeParticipantFromParty(final int partyId, final int participantId);

    @Query("SELECT contrib_comment, contrib_amount from ActiveParties where partyId=:partyId and participantId=:participantId")
    LiveData<List<Contribution>> getContributions(final int partyId, final int participantId);



}

File: app/src/main/java/com/partycalc/database/ActivePartiesDAO.kt
package com.partycalc.database

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface ActivePartiesDAO {
    @Insert
    fun insert(vararg item: ActiveParties): LongArray

    @Update
    fun update(vararg item: ActiveParties?)

    @Delete
    fun delete(vararg item: ActiveParties?)

    @Query("SELECT participant.* from activeparties " +
            "LEFT JOIN participant ON participant.id=activeparties.participantId " +
            "where partyId=:partyId")
    fun getParticipantsForParty(partyId: Int): LiveData<List<Participant>>

    @Query("delete from ActiveParties where participantId=:participantId and partyId=:partyId")
    fun removeParticipantFromParty(partyId: Int, participantId: Int)

    @Query("SELECT contrib_comment, contrib_amount from ActiveParties where partyId=:partyId and participantId=:participantId")
    fun getContributions(partyId: Int, participantId: Int): LiveData<List<Contribution>>
}